torch/csrc/utils/python_arg_parser.h

```diff
@@ -274,8 +274,7 @@ inline PythonArgs PythonArgParser::parse(PyObject* self, ParsedArgs<0>& dst) {
 
 inline bool PythonArgs::has_torch_function(){
   return !this->signature.overloaded_args.empty() ||
-      (torch::torch_function_enabled() &&
-       at::impl::PythonTorchFunctionTLS::get_mode());
+       at::impl::PythonTorchFunctionTLS::get_mode();
```

Comment:
hmmm, this is kind of weird now because now we're saying that torch function mode doesn't respect _disabled_torch_function_impl. Is this an argument for torch_function mode having a separate name from torch_function?

Reply:
Yes, it is confusing. The intended use case for having these be separate is to allow both "mode runs first, then subclass" and "subclass runs first, then mode", where the subclass is an "is-a" style subclass (like the default Tensor subclass handling). In this world, you need to be able to disable subclass dispatch without also disabling mode dispatch (because you want mode to handle things next).

I'm not sure having the names being different would be that helpful; a more descriptive name for 'DisableTorchFunction' might help.